main.tf

```diff
@@ -79,7 +79,11 @@ resource "azurerm_key_vault_access_policy" "main" {
   storage_permissions     = var.access_policies[count.index].storage_permissions
 }
 
-resource "azurerm_monitor_diagnostic_setting" "keyvault" {
+data "azurerm_monitor_diagnostic_categories" "default" {
+  resource_id = azurerm_key_vault.main.id
+}
+
+resource "azurerm_monitor_diagnostic_setting" "namespace" {
```

Comment:
@InsulaVentus ```resource "azurerm_monitor_diagnostic_setting" "namespace"``` should keep the same "keyvault" name to keep the terraform state consitent.